components/Auction/index.js

```diff
@@ -11,43 +11,25 @@ const WALLET_URL = `${BASE_API_URL}/wallet`;
 
 const HandleAuctions = () => {
   const [auctionsData, setAuctionsData] = useState([]);
-  const [userBid, setUserBid] = useState();
+  const [userBid, setUserBid] = useState(0);
   const [isUserLoggedIn, setIsUserLoggedIn] = useState(false);
   const [isLoading, setIsLoading] = useState(true);
-  const [userMoney, setUserMoney] = useState();
-
-  useEffect(() => {
-    fetchAndSetAuctions();
-  }, []);
-
-  useEffect(() => {
-    getUserWallet();
-  }, []);
-
-  useEffect(() => {
-    fetchSelfDetails()
-      .then((res) => {
-        if (res.status === 200) {
-          setIsUserLoggedIn(true);
-        }
-      })
-      .catch((err) => {
-        console.log('User is not logged in', err);
-      });
-  }, []);
+  const [userMoney, setUserMoney] = useState(0);
 
   const fetchAndSetAuctions = async () => {
     const response = await fetchData(AUCTIONS_URL);
-    const json = await response.json();
-    setAuctionsData(json.auctions);
+    if (!response) return null;
+    setAuctionsData(
+      JSON.parse(response && response.auctions ? response.auctions : 0) || 0
```

Comment:
You could use the new syntax here!
`JSON.parse(response.auctions ?? 0)`
It mostly won't need the fallback `|| 0` as well 😄